File: WinForms/VB/SchedulerMainDemo/Modules/Reporting/BaseModules/SingleReportPreviewControl.vb
Imports DevExpress.XtraScheduler.Reporting

Namespace DevExpress.XtraScheduler.Demos.Reporting

    Public Partial Class SingleReportPreviewControl
        Inherits ReportPreviewControlBase

        Private _report As XtraSchedulerReport

        Public Overridable ReadOnly Property Report As XtraSchedulerReport
            Get
                Return _report
            End Get
        End Property

        Protected Overridable Function CreateReportInstance() As XtraSchedulerReport
            Return Nothing
        End Function

        Protected Overrides Sub CreateReports()
            System.ComponentModel.DXDisplayNameAttribute.UseResourceManager = True
            If _report IsNot Nothing Then
                _report.Dispose()
            End If

            _report = CreateReportInstance()
        End Sub

        Protected Overrides Function GetActiveReport() As XtraSchedulerReport
            Return Report
        End Function
    End Class
End Namespace

File: WinForms/CS/SchedulerMainDemo/Modules/Reporting/BaseModules/SingleReportPreviewControl.cs
using System;
using DevExpress.XtraScheduler.Reporting;
using DevExpress.XtraReports.UserDesigner;

namespace DevExpress.XtraScheduler.Demos.Reporting {

    public partial class SingleReportPreviewControl : ReportPreviewControlBase {
        XtraSchedulerReport _report;
        public virtual XtraSchedulerReport Report { get { return _report; } }

        protected virtual XtraSchedulerReport CreateReportInstance() {
            return null;
        }
        protected override void CreateReports() {
            System.ComponentModel.DXDisplayNameAttribute.UseResourceManager = true;
            if (this._report != null) {
                this._report.Dispose();
            }
            this._report = CreateReportInstance();
        }
        protected override XtraSchedulerReport GetActiveReport() {
            return Report;
        }
        
	}

}

